File: middleware/src/dsl/Component.java
package dsl;

import java.util.HashMap;
import java.util.Map;

import Coordinate;

public class Component {
	protected String name;
	
	private Map<String,IntProperty> intProperties = new HashMap<String,IntProperty>();
	private Map<String,StringProperty> stringProperties = new HashMap<String,StringProperty>();
	private Map<String,DoubleProperty> doubleProperties = new HashMap<String,DoubleProperty>();
	private Component parent;
	
	public void setIntComponentProperty(String name, int value) {
		intProperties.put(name, new IntProperty(name, value));
	}
	
	public void setParent(Component parent) {
		this.parent = parent;
	}
	
	public void setDoubleComponentProperty(String name, double value) {
		doubleProperties.put(name, new DoubleProperty(name, value));
	}
	
	public void setStringComponentProperty(String name, String value) {
		stringProperties.put(name, new StringProperty(name,value));
	}
	
	public int getIntComponentProperty(String name) throws MissingProperty {
		IntProperty p = intProperties.get(name);
		if (p == null) {
			throw new MissingProperty(this,name);
		} else return p.getValue();
	}
	
	public String getStringComponentProperty(String name) throws MissingProperty {
		StringProperty p = stringProperties.get(name);
		if (p == null) {
			throw new MissingProperty(this,name);
		} else return p.getValue();
	}
	
	public double getDoubleComponentProperty(String name) throws MissingProperty {
		DoubleProperty p = doubleProperties.get(name);
		if (p == null) {
			throw new MissingProperty(this,name);
		} else return p.getValue();
	}

	public Component getParent() {
		return parent;
	}
}

File: middleware/dsl/mission/Component.py
from .IntProperty import IntProperty
from .StringProperty import StringProperty
from .DoubleProperty import DoubleProperty
from .CoordinateProperty import CoordinateProperty

class Component:

	def __init__(self):
		self.intProperties = {}
		self.stringProperties = {}
		self.doubleProperties = {}
		self.coordinateProperties = {}
	
	def set_int_component_property(self, name, value):
		self.intProperties[name] = IntProperty(name, value)
	
	def set_parent(self, parent):
		self.parent = parent
	
	def set_double_component_property(self, name, value):
		self.doubleProperties[name] =  DoubleProperty(name, value)
	
	def set_string_component_property(self, name, value):
		self.stringProperties[name] =  StringProperty(name,value)
	
	def set_coordinate_component_property(self, name, value):
		self.coordinateProperties[name] =  CoordinateProperty(name, value)

	def get_int_component_property(self, name):
		p = self.intProperties[name]
		if p != None:
			return p.getValue()
		return None
	
	def get_string_component_property(self, name):
		p = self.stringProperties[name]
		if p != None:
			return p.getValue()
		return None
	
	def get_double_component_property(self, name):
		p = self.doubleProperties[name]
		if p != None:
			return p.getValue()
		return None
	
	def get_coordinate_component_property(self, name):
		p = self.coordinateProperties[name]
		if p != None:
			return p.getValue()
		return None

	def get_parent(self):
		return self.parent
